contracts/BSCValidatorSet.sol

```diff
@@ -217,6 +232,83 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
   }
 
   /*********************** External Functions **************************/
+  /**
+   * @dev Update validator set method after fusion fork.
+   */
+  function updateValidatorSetV2(
+    address[] memory _consensusAddrs,
+    uint64[] memory _votingPowers,
+    bytes[] memory _voteAddrs
+  ) public onlyCoinbase onlyZeroGasPrice {
+    uint256 _length = _consensusAddrs.length;
+    Validator[] memory _validatorSet = new Validator[](_length);
+    for (uint256 i; i < _length; ++i) {
+      _validatorSet[i] = Validator({
+        consensusAddress: _consensusAddrs[i],
+        feeAddress: payable(address(0)),
+        BBCFeeAddress: address(0),
+        votingPower: _votingPowers[i],
+        jailed: false,
+        incoming: 0
+      });
+    }
+
+    // if staking channel is not closed, store the migrated validator set and return
+    if (ICrossChain(CROSS_CHAIN_CONTRACT_ADDR).registeredContractChannelMap(VALIDATOR_CONTRACT_ADDR, STAKING_CHANNELID)) {
+      uint256 newLength = _validatorSet.length;
+      if (newLength == 0) {
```

Comment:
There is a potential risk that: 
if only one validator migrate to BSC, if the validator is jailed, and causes `newLength` here, and `_tmpMigratedValidatorSet` will not update, so the jailed validator still can work.

I will suggest we discard the code here.

Reply:
Can you also check that will Beacon Chain still send cross chain package at breath block if there are no BSC validators on Beacon chain at all?